cmd/internal/apidump/apidump.go

```diff
@@ -121,9 +121,16 @@ func apidumpRunInternal(cmd *cobra.Command, _ []string) error {
 		return errors.Wrap(err, "failed to load plugins")
 	}
 
-	// Check that exactly one of --project or --collection is specified.
+	// override the project id if the environment variable is set
+	if envProjectID := os.Getenv("POSTMAN_INSIGHTS_PROJECT_ID"); envProjectID != "" {
+		projectID = envProjectID
+	}
+
+	// Check that exactly one of --project or --collection is specified, or POSTMAN_INSIGHTS_PROJECT_ID was set.
+	// If not, we wll loop indefinitely to avoid boot loops.
 	if projectID == "" && postmanCollectionID == "" {
-		return errors.New("exactly one of --project or --collection must be specified")
+		printer.Stdout.Infof("This process will not exit, to avoid boot loops. Please correct the command line flags or environment and retry.\n")
```

Comment:
This log message doesn't say what the problem is.

Needs to say something like "Please specify a project ID with the --project flag or the POSTMAN_INSIGHTS_PROJECT_ID environment variable."